clinica/pipelines/dwi_preprocessing_using_t1/dwi_preprocessing_using_t1_utils.py

```diff
@@ -240,87 +243,213 @@ def print_end_pipeline(image_id, final_file):
         )
+    if nb_b0s == 1:
+        return extracted_b0
+    registered_b0s = register_b0(
+        nb_b0s,
+        extracted_b0,
+        working_directory=working_directory,
+    )
+    out_reference_b0 = compute_average_b0(registered_b0s, squeeze=False)
+    registered_b0_file_name = extracted_b0.with_name("reference_b0_volume.nii.gz")
+    shutil.copy(out_reference_b0, registered_b0_file_name)
+    if clean_working_dir:
+        shutil.rmtree(working_directory)
+
+    return registered_b0_file_name
+
+
+def configure_working_directory(
+    dwi_filename: Path,
+    working_directory: Optional[Path] = None,
+) -> Path:
+    """Configures a temporary working directory for writing the output files of
+    the b0 co-registration.
+
+    Parameters
+    ----------
+    dwi_filename : Path
+        Path to DWI file. This is used to create the name of the folder.
+
+    working_directory : Path, optional
+        The folder to be used if provided by the user.
+        If None, then create a temporary folder to work in.
+
+    Returns
+    -------
+    str:
+        The configured working directory.
+    """
+    import hashlib
+    import tempfile
+    from pathlib import Path
 
-    # Merge datasets such that bval(DWI) = (0 b1 ... bn)
-    [out_b0_dwi_merge, out_updated_bval, out_updated_bvec] = insert_b0_into_dwi(
-        in_b0=out_reference_b0,
-        in_dwi=out_split_dwi,
-        in_bval=out_split_bval,
-        in_bvec=out_split_bvec,
+    working_directory = working_directory or tempfile.mkdtemp()
+    working_directory = (
+        Path(working_directory) / hashlib.md5(str(dwi_filename).encode()).hexdigest()
     )
+    working_directory.mkdir(parents=True)
 
-    return out_reference_b0, out_b0_dwi_merge, out_updated_bval, out_updated_bvec
+    return working_directory
+
+
+def register_b0(
+    nb_b0s: int,
+    extracted_b0_filename: Path,
+    working_directory: Path,
+) -> Path:
+    """Run the FSL pipeline 'b0_flirt_pipeline' in order to co-register the b0 images.
+
+    This function is a simple wrapper around the b0_flirt_pipeline which configures it,
+    runs it, and returns the path to the merged file of interest.
+
+    Parameters
+    ----------
+    nb_b0s : int
+        The number of B0 volumes in the dataset.
+
+    extracted_b0_filename : Path
+        The extracted B0 volumes to co-register.
+
+    working_directory : Path
+        The working directory in which the pipeline will write intermediary files.
+
+    Returns
+    -------
+    Path:
+        The path to the nifti image containing the co-registered B0 volumes.
+        If the pipeline ran successfully, this file should be located in :
+        working_directory / b0_coregistration / concat_ref_moving / merged_files.nii.gz
+    """
+    from clinica.pipelines.dwi_preprocessing_using_t1.dwi_preprocessing_using_t1_workflows import (
+        b0_flirt_pipeline,
+    )
+
```

Comment:
they come from the automatic formatting.
This makes me think that we should clean these module names a bit when we get the chance. 
The file could just be named `utils.py` since it is in the `dwi_preprocessing_using_t1` folder.